cmd/kubectl_token.go

```diff
@@ -0,0 +1,546 @@
+package cmd
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"net/http"
+	url2 "net/url"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/rancher/norman/types/convert"
+	managementClient "github.com/rancher/types/client/management/v3"
+	"github.com/urfave/cli"
+	"golang.org/x/crypto/ssh/terminal"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
+)
+
+type LoginInput struct {
+	server       string
+	userID       string
+	clusterID    string
+	authProvider string
+	caCerts      string
+	skipVerify   bool
+}
+
+const (
+	kubeConfigCache = "/.cache/token"
+	cachedFileExt   = ".json"
+	authProviderURL = "%s/v3-public/authProviders"
+	authTokenURL    = "%sv3-public/authtokens/%s"
+)
+
+var samlProviders = map[string]bool{
+	"pingProvider":       true,
+	"adfsProvider":       true,
+	"keycloakProvider":   true,
+	"oktaProvider":       true,
+	"shibbolethProvider": true,
+}
+
+func CredentialCommand() cli.Command {
+	return cli.Command{
+		Name:        "token",
+		Usage:       "Authenticate and generate new kubeconfig token",
+		Description: "todo",
+		Action:      runCredential,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "server",
```

Comment:
does the cli already not ask for this?

Reply:
yep, it's the same. 
cli asks for it, but since this is invoked on `kubectl` calls, it's not necessary that all users would have configured their cli context before that so server gets passed from the kubeconfig generate itself